pkg/scheduler/factory.go

```diff
@@ -268,138 +267,58 @@ func (c *Configurator) createFromConfig(policy schedulerapi.Policy) (*Scheduler,
 
 	klog.V(2).Infof("Creating scheduler with fit predicates '%v' and priority functions '%v'", predicateKeys, priorityKeys)
 
-	pluginsForPredicates, pluginConfigForPredicates, err := getPredicateConfigs(predicateKeys, lr, args)
-	if err != nil {
-		return nil, err
-	}
-
-	pluginsForPriorities, pluginConfigForPriorities, err := getPriorityConfigs(priorityKeys, lr, args)
-	if err != nil {
-		return nil, err
-	}
 	// Combine all framework configurations. If this results in any duplication, framework
 	// instantiation should fail.
-	var defPlugins schedulerapi.Plugins
+
 	// "PrioritySort" and "DefaultBinder" were neither predicates nor priorities
 	// before. We add them by default.
-	defPlugins.Append(&schedulerapi.Plugins{
+	plugins := schedulerapi.Plugins{
 		QueueSort: &schedulerapi.PluginSet{
 			Enabled: []schedulerapi.Plugin{{Name: queuesort.Name}},
 		},
 		Bind: &schedulerapi.PluginSet{
 			Enabled: []schedulerapi.Plugin{{Name: defaultbinder.Name}},
 		},
-	})
-	defPlugins.Append(pluginsForPredicates)
-	defPlugins.Append(pluginsForPriorities)
-	defPluginConfig, err := mergePluginConfigsFromPolicy(pluginConfigForPredicates, pluginConfigForPriorities)
+	}
+	var pluginConfig []schedulerapi.PluginConfig
+	var err error
+	pluginConfig, err = lr.GetPredicateConfigs(predicateKeys, args, &plugins, pluginConfig)
+	if err != nil {
+		return nil, err
+	}
+	pluginConfig, err = lr.GetPriorityConfigs(priorityKeys, args, &plugins, pluginConfig)
+	if err != nil {
+		return nil, err
+	}
+	pluginConfig, err = dedupPluginConfigs(pluginConfig)
 	if err != nil {
 		return nil, err
 	}
 	for i := range c.profiles {
 		prof := &c.profiles[i]
-		// Plugins are empty when using Policy.
-		prof.Plugins = &schedulerapi.Plugins{}
-		prof.Plugins.Append(&defPlugins)
```

Comment:
out of curiosity why did we have to originally do this (append instead of simply assign)? :)

Reply:
Two reasons (?).

1. Originally we were merging Policy and Plugins
2. Append initializes the PluginSets. I guess this might cause issues. I'll revert, but opened #95190